cfme/control/explorer/alerts.py

```diff
@@ -304,6 +306,30 @@ def create(self, description, severity=None, active=None, based_on=None, evaluat
         view.flash.assert_success_message('Alert "{}" was added'.format(alert.description))
         return alert
 
+    def all(self):
+        # get alerts via reading 'All Alerts table'
+        view  = navigate_to(self.appliance.collections.alerts,'All')
+        try:
+            alert_table = view.table
+            alerts = [
+                self.instantiate(
+                      alert.get('Description'),
```

Comment:
I think you are missing the key here:
```description= alert.get('Description')```

Reply:
This is because the description does not have a default value, it is an argument to the instantiate method. c.f. line 295 of `cfme/control/explorer/alerts.py`

```
        alert = self.instantiate(description, severity=severity, active=active, based_on=based_on,
            evaluate=evaluate, driving_event=driving_event,
            notification_frequency=notification_frequency, snmp_trap=snmp_trap, emails=emails,
            timeline_event=timeline_event, mgmt_event=mgmt_event)
```

Reply:
After checking, it appears both work and are equivalent. Is it preferred to include the key?

Reply:
@john-dupuy It is good practice to include the key. It helps the understandability of the call.